[FCA-15] Change spinner to recycler view on main screen

matxowy opened the pull request:
- Implemented RecyclerView on main screen instead of spinner
- Improved code

matxowy committed:
[FCA-15]

- Added category item and change to recyclerview

matxowy committed:
[FCA-15]

- Implemented RecyclerView on main screen instead of spinner
- Improved code

matxowy committed:
[FCA-15]

- Added DatabaseView
- Migrated database to version 2
- Removed unnecessary increment use case

msniezek-Sii commented on app/src/main/java/com/matxowy/flashcardsapp/data/db/dao/CategoryDao.kt:
```diff
@@ -7,13 +7,17 @@ import androidx.room.OnConflictStrategy
     fun getCategories(): Flow<List<Category>>
 
+    @Query("SELECT * FROM CategoryDetail ORDER BY id")
+    fun getCategoriesWithDetails(): Flow<List<CategoryDetail>>
```
Database view możesz traktować jak tabelę i może mieć swoje własne Dao

matxowy replied:
Rozdzielone ☺️ 
Miałem jakieś problemy z apką przy automigracji dlatego dodatkowo dla pewności, że będzie wszystko działać zrobiłem manualną migrację 😄

matxowy committed:
[FCA-15]

- Added separate dao for CategoryDetail view
- Added manual migration due to problems with auto migrations
